llvm/lib/CodeGen/InlineSpiller.cpp

```diff
@@ -946,6 +946,9 @@ foldMemoryOperand(ArrayRef<std::pair<MachineInstr *, unsigned>> Ops,
     if (MO.isUse() && !MO.readsReg() && !MO.isTied())
       continue;
 
+    if (MI->isCopy() && MI->getOperand(1).isUndef())
```

Comment:
This shouldn't have gotten this far in the first place? We have a number of places already deleting copies of undef

Reply:
Oh, I now see in the other PR, this is from hacking up the failed ranges. Can you add the additional context to the commit message? There also might still be a better way to do this at the failure point

Reply:
Copy shouldn't be special, the undef read case is already handled above here by the !readsReg check?

Reply:
> Oh, I now see in the other PR, this is from hacking up the failed ranges. 

Yeah -- the test case is probably misleading, since it's using the MIR mid RA (i.e. after cleanedFailedVReg has created its version of the liveness)

Reply:
> There also might still be a better way to do this at the failure point

I guess instead of creating undef copies we can just create implicit_defs. Seems like sort of an ad-hoc requirement, but cleanupFailedVReg seems like a hack anyways.

> undef read case is already handled above here by the !readsReg check?

Do you mean the dest of an undef copy should be !readsReg ? This is checking conditions of the register we plan to spill / fold, so, in this case, the dest of the copy.

Reply:
In the latest, I've changed it such that cleanupFailedVreg produces IMPLICIT_DEF instead of undef copies